File: kyu_8/039_Beginner_Reduce_but_Grow/javascript/js/main.js
// solution 1
function grow(x) {
  return x.reduce((acc, curr) => acc * curr, 1);
}

console.log(grow([1, 2, 3, 4]));

// solution 2
function grow(x) {
  let result = 1;

  for (let i = 0; i < x.length; i++) 
    result *= x[i];

  return result;
}

console.log(grow([1, 2, 3, 4]));

File: kyu_8/039_Beginner_Reduce_but_Grow/c/main.c
#include <stddef.h>
#include <stdio.h>


int grow(size_t size, const int arr[size]) {
  	int result = 1;
	size_t i;

	for (i = 0; i < size; i++)
		result *= arr[i];

	return (result);
}

int main(void)
{
	size_t size = 4;
	int array[] = {1, 2, 3, 4};

	printf("%d\n", grow(size, array)); /* 24 */

	return (0);
}
